master/docs/developer/master-worker-msgpack.rst

```diff
@@ -0,0 +1,1290 @@
+Master-Worker connection with MessagePack over WebSocket protocol
+.. note::
+    This is experimental protocol.
+=================================================================
+
+Messages between master and worker are sent using WebSocket protocol in both directions.
+Data to be sent conceptually is a dictionary and is encoded using MessagePack.
+One such encoded dictionary corresponds to one WebSocket message.
+
+A message can be either a request or a response.
+Request message is sent when one counter party wants another one to perform an action.
```

Comment:
counter party? never heard of this term. should we just use 'side'?

Reply:
Fixed.